docs/source/_src_user_guide/state_propagation/propagation_setup/acceleration_models/thrust.rst

```diff
@@ -0,0 +1,301 @@
+
+         .. literalinclude:: /_src_snippets/simulation/propagation_setup/thrust/thrust_direction_from_state_guidance.py
+            :language: python
+
+      .. tab:: C++
+
+         .. literalinclude:: /_src_snippets/simulation/propagation_setup/thrust/thrust_direction_from_state_guidance.cpp
+            :language: cp
+
+Custom thrust direction settings
+================================
+
+For a generalized thrust direction guidance, the thrust direction can be defined as an arbitrary function of time. This allows a broad range of options to be defined, at the expense of increased complexity — somehow this thrust direction needs to be manually defined.
+
+A custom thrust direction can be defined **in the inertial frame** using the following:
+
+   .. tabs::
+
+      .. tab:: Python
+
+         The Tudat(Py) API docs give more details on the :func:`~tudatpy.numerical_simulation.propagation_setup.thrust.custom_thrust_direction` function.
+
+         .. literalinclude:: /_src_snippets/simulation/propagation_setup/thrust/custom_thrust_direction.py
+            :language: python
+
+      .. tab:: C++
+
+         .. literalinclude:: /_src_snippets/simulation/propagation_setup/thrust/custom_thrust_direction.cpp
+            :language: cp
+
+.. warning:: When using this option, the inertial to body-fixed rotation cannot be unambiguously defined. If you require this rotation (for instance when you also incorporate aerodynamic forces), the :func:`~tudatpy.numerical_simulation.propagation_setup.thrust.custom_thrust_orientation` option should be used instead.
+
+.. warning:: The direction vector that is being returned by the custom function should be a unit vector.
+
+Custom thrust orientation settings
+==================================
+
+Thrust orientation settings can also be created trough a custom function that returns this time not the direction but the orientation of the thrust.
+
+This thrust orientation needs to be provided through a rotation matrix representing the rotation from body-fixed thrust direction to the inertial thrust direction.
+
+   .. tabs::
+
+      .. tab:: Python
+
+         The Tudat(Py) API docs give more details on the :func:`~tudatpy.numerical_simulation.propagation_setup.thrust.custom_thrust_orientation` function.
+
+         .. literalinclude:: /_src_snippets/simulation/propagation_setup/thrust/custom_thrust_orientation.py
+            :language: python
+
+      .. tab:: C++
+
+         .. literalinclude:: /_src_snippets/simulation/propagation_setup/thrust/custom_thrust_orientation.cpp
+            :language: cp
+
+Thrust direction from existing orientation
+==========================================
+
+The orientation of the vehicle is in some cases already defined. This could be thanks to aerodynamic guidance or to the propagation of rotational dynamics. In this context, the thrust direction can be computed from the body-fixed direction. 
```

Comment:
> In this context, the thrust direction can be computed from the body-fixed direction.

Do you mean "body-fixed orientation"? I am not an expert about thrust in tudat and from reading through this page I get a bit confused between orientation and direction. Let's try to be careful in using the correct terms (perhaps this is already the case, not sure) to avoid confusion for the users

Reply:
I do, indeed.
I think it's normal that you are confused, to be honest, the current implementation is confusing I would say (though it does work).
Referring to my previous comment, we do seek to make this less confusing in the future.

Reply:
And I totally agree that, being myself still partially confused by this orientation/direction definition (especially combined with aerodynamic orientation or rotational dynamics), I should make even more of an effort to not transmit this confusion to users 🙂